contracts/src/Teleporter/ReceiptQueue.sol

```diff
@@ -66,32 +65,53 @@ contract ReceiptQueue {
      * - `msg.sender` must be the owner.
      * - The queue must be non-empty.
      */
-    function dequeue()
-        external
+    function dequeue(
+        TeleporterMessageReceiptQueue storage queue
+    )
+        internal
         returns (TeleporterMessageReceipt memory result)
     {
-        if (msg.sender != owner) {
-            revert Unauthorized();
-        }
+        uint256 first_ = queue.first;
+        if (queue.last == first_) revert EmptyQueue();
+        result = queue.data[first_];
+        delete queue.data[first_];
+        queue.first = first_ + 1;
```

Comment:
Is it worthwhile to reset first and last to their initial value (0) if they are ever equal after a dequeue?